### PR TITLE
perf(frontend) : Setup redux toolkit with persist

### PR DESCRIPTION
**Pull request by Arielpetit**
Please review this PR @stephane-segning @ValantineSuh @Ngha-Boris

### CODE REVIEW

**Review by ValantineSuh — APPROVED** [review]
Good!

File: src/app.tsx [hunk 1]
```diff
@@ -1,18 +1,28 @@
...
-        <Route path="/" element={<Homepage />} />
-        <Route path="/login" element={<UserLogin />} />
-      </Routes>
-    </Router>
+    <Provider store={store}>
+      <Router>
+        <div>
+          <LoginButton />
+          <LogoutButton />
+        </div>
```
Comment (Ngha-Boris) [inline src/app.tsx:20]: The LoginButton and LogoutButton are rendered outside of routes in app.tsx. This means they'll be visible on all pages.
Please @Arielpetit, consider conditionally rendering the buttons based on the isLoggedIn state.

**Review by Ngha-Boris — CHANGES_REQUESTED** [review]

**Review by stephane-segning — CHANGES_REQUESTED** [review]
You integrated redux, not redux toolkit. Please read it again [here](https://redux-toolkit.js.org/introduction/getting-started) and [here](https://redux-toolkit.js.org/usage/usage-with-typescript). Read first before the single implementation please.

File: package.json [hunk 2]
```diff
@@ -15,11 +15,15 @@
     "stylelint:actions": "stylelint --fix \"src/**/*.{css,scss}\""
   },
   "dependencies": {
+    "@types/redux-persist": "^4.3.1",
```
[the remark this answers is not in the document]
Comment (Arielpetit) [reply package.json]: Done!!!

File: src/components/LoginButton.tsx
[the remark this answers is not in the document]
Comment (Arielpetit) [reply src/components/LoginButton.tsx:1]: Done

File: src/store/actions.ts
[the remark this answers is not in the document]
Comment (Arielpetit) [reply src/store/actions.ts:1]: Done

File: src/store/reducers.ts
[the remark this answers is not in the document]
Comment (Arielpetit) [reply src/store/reducers.ts:1]: Done

**Comment by Arielpetit** [comment]
Hey please @stephane-segning and @Ngha-Boris review this again

File: .husky/pre-commit
Comment (Ngha-Boris) [inline .husky/pre-commit:1]: Hello @Arielpetit what is this file for

File: package.json [hunk 3]
```diff
@@ -29,6 +32,8 @@
     "@eslint/eslintrc": "^3.0.2",
     "@types/react": "^18.2.66",
     "@types/react-dom": "^18.2.22",
+    "@reduxjs/toolkit": "^2.2.3",
```
Comment (stephane-segning) [inline package.json:35]: This should be in dep, not dev dep

File: src/components/home-page.tsx
Comment (stephane-segning) [inline src/components/home-page.tsx:1]: Cancel all these changes

**Review by stephane-segning — CHANGES_REQUESTED** [review]
Very good improvement. Now, address these points please

File: src/components/home-page.tsx
Comment (Arielpetit) [reply src/components/home-page.tsx:1]: Done

File: package.json (hunk 3 above)
Comment (Arielpetit) [reply package.json]: DOne

File: .husky/pre-commit
Comment (stephane-segning) [inline .husky/pre-commit:1]: Cancel this change

**Review by stephane-segning — CHANGES_REQUESTED** [review]

File: .husky/pre-commit
Comment (ValantineSuh) [inline .husky/pre-commit:1]: Kindly cancel the change requested by Sir Stephane.

**Pull request closed without merging**